### DISCUSSION TITLE
blobfuse2 difference between --attr-cache-timeout and --attr-timeout

### DISCUSSION THREAD

**Discussion by Mmdixon** [Azure/azure-storage-fuse#940] [Q&A] [answered]
If `--attr-timeout` (libfuse.attribute-expiration-sec) is the libfuse option for "time kernel can cache inode attributes". Then what is `--attr-cache-timeout` (attr_cache.timeout-sec) that states "The timeout for the attribute cache entries"? What do they do differently and why do they have different timeouts?

**Comment by bhansalivikas** [comment, marked as answer]
--attr-timeout : how long kernel can cache attributes of a given file
--attr-cache-timeout : how long blobfuse attr-cache component can cache the attribute of a given file. If this timeout is greater than kernel timeout even when kernel asks us to refresh the attribute, we can still use our cache instead of asking the storage. This caching is little intelligent compared to kernel one as we keep updating the size and last modified time of a file in local cache as and when file operations are performed. This allows us to reduce GetProperties() call done to storage which are very frequently done by kernel to get metadata of a file.
Also, one major difference about blobfuse caching attribute and kernel caching it is kernel caches only when it does a getStat() call on a file while blobfuse caches it whenever you do a list call on a directory also. As part of list call, we get metadata of all the files of given directory and blobfuse caches all that meta. This allows us to reduce our GetProperties() call significantly and save a lot on network calls.

**Comment by ArulkumarPonnusamy** [comment]
if i change the file content in storage account, the changes are not reflecting in mount path file. Please suggest
